src/main/proto/upayload.proto

```diff
@@ -38,16 +38,17 @@
 // the pointer address to the data. If data is passed by value, the data is stored in the
 // value field. The length field is required if data is passed by reference. The format
 message UPayload {
-    oneof data {
-        fixed64 reference = 1; // Data is passed by reference so this is the pointer address
-        bytes value = 2;        // Data is passed by value so value contains a copy of the data
-    }
-
+    // Data is passed by value so value contains a copy of the data
+    bytes value = 1;       
```

Comment:
Is this is not optional then we will need to always pass by value (not your point AFAIK)